### PR TITLE
Move part of PluginContext to mol-plugin-ui

### PR DESCRIPTION
**Pull request by dsehnal**
Moves all React dependencies from ``mol-plugin`` to ``mol-plugin-ui`` by introducing ``PluginUIContext``.

### CODE REVIEW

**Comment by arose** [comment]
Great, I will try it tomorrow.

File: src/mol-plugin/context.ts [hunk 1]
```diff
@@ -176,9 +174,8 @@ export class PluginContext {
 
     readonly customModelProperties = new CustomProperty.Registry<Model>();
     readonly customStructureProperties = new CustomProperty.Registry<Structure>();
-    readonly customParamEditors = new Map<string, StateTransformParameters.Class>();
 
-    readonly customStructureControls = new Map<string, { new(): PluginUIComponent<any, any, any> }>();
+    readonly customStructureControls = new Map<string, { new(): any /* contructible react components */ }>();
```
Comment (arose) [inline src/mol-plugin/context.ts:178]: typo: constructible

**Review by arose — COMMENTED** [review]
This works well, no issues with including react or UI related code. Thanks!. The only aspects that's annoying is that you now have register Actions manually if you don't use the UI version.

**Comment by dsehnal** [comment]
Ok I will move the actions back to the original spec, there is no UI there.

**Pull request merged**